projects/aca-content/src/lib/components/search/search-input/search-input.component.spec.ts

```diff
@@ -1,62 +1,116 @@
+    const error = fixture.debugElement.query(By.directive(MatError));
+    return error?.nativeElement.textContent.trim();
+  }
 
-    // (component as any).searchInputControl = mockControl;
+  function openSearchContainer(): void {
+    const menuButton = fixture.debugElement.query(By.css('.app-search-container'));
+    menuButton?.nativeElement.click();
     fixture.detectChanges();
+  }
+
+  it('should show required error when field is empty and touched', () => {
+    openSearchContainer();
+    component.searchInputControl.searchFieldFormControl.setValue('');
+    component.searchInputControl.searchFieldFormControl.markAsTouched();
+    fixture.detectChanges();
+    expect(getFirstError()).toBe('SEARCH.INPUT.REQUIRED');
   });
 
-  describe('Validation Behavior', () => {
-    function getFirstError(): string {
-      const error = fixture.debugElement.query(By.directive(MatError));
-      return error?.nativeElement.textContent.trim();
-    }
-
-    it('should show required error when field is empty and touched', () => {
-      component.searchInputControl.searchFieldFormControl.setValue('');
-      component.searchInputControl.searchFieldFormControl.markAsTouched();
-      fixture.detectChanges();
-
-      const error = getFirstError();
-      fixture.detectChanges();
-      expect(error).toBe('SEARCH.INPUT.REQUIRED');
-    });
-
-    // it('should not show error when field has value', () => {
-    //   mockControl.searchFieldFormControl.setValue('test');
-    //   mockControl.searchFieldFormControl.markAsTouched();
-    //   fixture.detectChanges();
-
-    //   const error = fixture.debugElement.query(By.directive(MatError));
-    //   expect(error).toBeNull();
-    // });
-
-    // it('should not show error when field is untouched', () => {
-    //   mockControl.searchFieldFormControl.setValue('');
-    //   mockControl.searchFieldFormControl.markAsUntouched();
-    //   fixture.detectChanges();
-
-    //   const error = fixture.debugElement.query(By.directive(MatError));
-    //   expect(error).toBeNull();
-    // });
+  it('should not show error when field has value', () => {
+    openSearchContainer();
+    component.searchInputControl.searchFieldFormControl.setValue('test');
+    component.searchInputControl.searchFieldFormControl.markAsTouched();
+    fixture.detectChanges();
+    const error = fixture.debugElement.query(By.directive(MatError));
+    expect(error).toBeNull();
+  });
+
+  it('should not show error when field is untouched', () => {
+    openSearchContainer();
+    component.searchInputControl.searchFieldFormControl.setValue('');
+    component.searchInputControl.searchFieldFormControl.markAsUntouched();
+    fixture.detectChanges();
+    const error = fixture.debugElement.query(By.directive(MatError));
+    expect(error).toBeNull();
+  });
+
+  it('should dispatch SearchByTermAction when libraries are checked and term is new', () => {
+    spyOn(component as any, 'isLibrariesChecked').and.returnValue(true);
```

Comment:
Don't spy on component's functions - you can spy on only things which are outside component like services. We can't spyOn them because it will change behavior of those functions. Test component based on those functions without spying them